packages/core/src/core/turn.ts

```diff
@@ -216,6 +216,41 @@ export class Turn {
             yield event;
           }
         }
+
+        if (resp.usageMetadata) {
+          this.lastUsageMetadata =
+            resp.usageMetadata as GenerateContentResponseUsageMetadata;
+        }
+
+        // Check if response was truncated or stopped for various reasons
+        const finishReason = resp.candidates?.[0]?.finishReason;
+        
+        const finishReasonMessages: Record<string, string> = {
```

Comment:
For all these would you mind migrating to the official `FinishReason.MAX_TOENS` etc.